src/main/java/org/icgc/argo/program_service/services/EgoService.java

```diff
@@ -390,10 +401,116 @@ public static GroupName createProgramGroupName(String name, UserRole role){
   }
 
 
+  public void updateUserRole(@NonNull UUID userId, @NonNull UUID programId, @NonNull UserRole role) throws NotFoundException, RuntimeException {
+    getUserById(userId);
+    val program = findProgramById(programId);
+    val shortname = program.getShortName();
+
+    val groups = getGroupsFromUser(userId);
+    if(groups.isEmpty()){
+      throw new NotFoundException(String.format("No groups found for user id %s.", userId));
+    }
+    // determine the current group
+    for(val group : groups){
+      if(group.getName().toLowerCase().contains(shortname.toLowerCase())) {
+        if(!isSameRole(role, group.getName())){
+          removeUserFromCurrentGroup(group, userId);
+        } else {
+          throw new RuntimeException(format("Cannot update user role %s, new role is the same as current role.", role.toString()));
```

Comment:
instead of throwing an exception, i would just log this as a warning and thus make it idempotent.